### PR TITLE
Add Scantist Vulnerability Scan Parser

### PR DESCRIPTION
**Pull request by mohcer**
https://scantist.com

### CODE REVIEW

File: dojo/unittests/test_scantist_parser.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    def test_parse_file_with_one_vuln_has_one_finding(self):
+        testfile = open("dojo/unittests/scans/scantist/scantist-one-vuln.json")
+        parser = ScantistJSONParser(testfile, Test())
+        self.assertEqual(1, len(parser.items))
+
+    def test_parse_file_with_multiple_vuln_has_multiple_findings(self):
+        testfile = open("dojo/unittests/scans/scantist/scantist-many-vuln.json")
+        parser = ScantistJSONParser(testfile, Test())
+        self.assertTrue(len(parser.items) > 2)
```
Comment (damiencarol) [inline dojo/unittests/test_scantist_parser.py:25]: Could you add more tests on the Findings? It's good to show that your parser supports a lot of this like for ex component name and version? (not mandatory)

File: dojo/tools/scantist/parser.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+            finding = Finding(
+                title=title,
+                test=test,
+                description=description,
+                severity=severity,
+                cve=cve,
+                cwe=vuln.get('cwe'),
+                mitigation=mitigation,
+                numerical_severity=Finding.get_numerical_severity(
```
Comment (damiencarol) [inline dojo/tools/scantist/parser.py:60]: Could you reuse the severity value?
```suggestion
                numerical_severity=Finding.get_numerical_severity(severity)
```

File: dojo/tools/scantist/parser.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            severity = vuln.get("Score", "N/A")
+            score = vuln.get("cvss_v3_score", "N/A")
+
+            mitigation = vuln.get("Patched Version")
+
+            finding = Finding(
+                title=title,
+                test=test,
+                description=description,
+                severity=severity,
```
Comment (damiencarol) [inline dojo/tools/scantist/parser.py:56]: There will be a pb with this one. If you don't have data for severity, the value "N/A" will produce exceptions when saving the Finding.

File: dojo/tools/scantist/parser.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                cve=cve,
+                cwe=vuln.get('cwe'),
```
Comment (damiencarol) [inline dojo/tools/scantist/parser.py:58]: `cwe` attribute is an integer. There is something to modify here

**Review by damiencarol — CHANGES_REQUESTED** [review]
Good first version but some parts need modifications.

File: dojo/tools/scantist/parser.py (hunk 3 above)
Comment (mohcer) [reply dojo/tools/scantist/parser.py:56]: From the look of it 
```
        if severity == 'Critical':
            return 'S0'
        elif severity == 'High':
            return 'S1'
        elif severity == 'Medium':
            return 'S2'
        elif severity == 'Low':
            return 'S3'
        elif severity == 'Info':
            return 'S4'
        else:
            return 'S5'
 ```
 
 should it return S5? is S5 unknown severity something of this sort?

File: .github/renovate.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
     "config:base"
   ],
   "dependencyDashboard": true,
-  "dependencyDashboardApproval": false,
+  "dependencyDashboardApproval": true,
```
Comment (damiencarol) [inline .github/renovate.json]:
```suggestion
  "dependencyDashboardApproval": false,
```

**Review by damiencarol — CHANGES_REQUESTED** [review]
One minor comment but LGTM

File: .github/renovate.json (hunk 5 above)
Comment (mohcer) [reply .github/renovate.json]: This change was undone in the last commit ..should it apply the suggestion again?

**Review by damiencarol — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**